pkg/registry/generic/etcd/etcd_test.go

```diff
@@ -137,88 +132,42 @@ func TestEtcdList(t *testing.T) {
 		Spec:       api.PodSpec{NodeName: "machine"},
 	}
 
-	singleElemListResp := &etcd.Response{
-		Node: &etcd.Node{
-			Value: runtime.EncodeOrDie(testapi.Default.Codec(), podA),
-		},
-	}
-	normalListResp := &etcd.Response{
-		Node: &etcd.Node{
-			Nodes: []*etcd.Node{
-				{Value: runtime.EncodeOrDie(testapi.Default.Codec(), podA)},
-				{Value: runtime.EncodeOrDie(testapi.Default.Codec(), podB)},
-			},
-		},
-	}
-
 	testContext := api.WithNamespace(api.NewContext(), "test")
 	noNamespaceContext := api.NewContext()
 
 	table := map[string]struct {
-		in      tools.EtcdResponseWithError
+		in      *api.PodList
 		m       generic.Matcher
 		out     runtime.Object
 		context api.Context
-		succeed bool
 	}{
-		"empty": {
-			in: tools.EtcdResponseWithError{
-				R: &etcd.Response{
-					Node: &etcd.Node{
-						Nodes: []*etcd.Node{},
-					},
-				},
-				E: nil,
-			},
-			m:       everythingMatcher{},
-			out:     &api.PodList{Items: []api.Pod{}},
-			succeed: true,
-		},
 		"notFound": {
-			in: tools.EtcdResponseWithError{
-				R: &etcd.Response{},
-				E: tools.EtcdErrorNotFound,
-			},
-			m:       everythingMatcher{},
-			out:     &api.PodList{Items: []api.Pod{}},
-			succeed: true,
+			in:  nil,
+			m:   everythingMatcher{},
+			out: &api.PodList{Items: []api.Pod{}},
 		},
 		"normal": {
-			in: tools.EtcdResponseWithError{
-				R: normalListResp,
-				E: nil,
-			},
-			m:       everythingMatcher{},
-			out:     &api.PodList{Items: []api.Pod{*podA, *podB}},
-			succeed: true,
+			// order here matters b/c return list will be sorted,
+			// and we should compare against the modified input stream
+			in:  &api.PodList{Items: []api.Pod{*podB, *podA}},
+			m:   everythingMatcher{},
+			out: nil,
```

Comment:
I cleaned up the function a bit for readability, let me know if it's easier to follow.